packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -0,0 +1,367 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'theme.dart';
+
+// Slides the page upwards and fades it in, starting from 1/4 screen
+// below the top.
+class _FadeUpwardsPageTransition extends StatelessWidget {
+  _FadeUpwardsPageTransition({
+    Key key,
+    @required Animation<double> routeAnimation, // The route's linear 0.0 - 1.0 animation.
+    @required this.child,
+  }) : _positionAnimation = routeAnimation.drive(_bottomUpTween.chain(_fastOutSlowInTween)),
+       _opacityAnimation = routeAnimation.drive(_easeInTween),
+       super(key: key);
+
+  // Fractional offset from 1/4 screen below the top to fully on screen.
+  static final Tween<Offset> _bottomUpTween = Tween<Offset>(
+    begin: const Offset(0.0, 0.25),
+    end: Offset.zero,
+  );
+  static final Animatable<double> _fastOutSlowInTween = CurveTween(curve: Curves.fastOutSlowIn);
+  static final Animatable<double> _easeInTween = CurveTween(curve: Curves.easeIn);
+
+  final Animation<Offset> _positionAnimation;
+  final Animation<double> _opacityAnimation;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return SlideTransition(
+      position: _positionAnimation,
+      // TODO(ianh): tell the transform to be un-transformed for hit testing
+      child: FadeTransition(
+        opacity: _opacityAnimation,
+        child: child,
+      ),
+    );
+  }
+}
+
+// This transition is intended to match the default for Android P.
```

Comment:
Since this is `//` internal documentation, I included the motion descriptions with the Tweens that define them.

The corresponding (public) PageTransitionBuilder does explain the motion in its class dartdoc.